### PR TITLE
Add `com.iexec.commons.poco.eip712` package

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

File: src/main/java/com/iexec/commons/poco/eip712/EIP712Domain.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the {@code EIP712Domain} part an EIP-712 compliant data structure.
```
Comment (mcornaton) [inline src/main/java/com/iexec/commons/poco/eip712/EIP712Domain.java:32]:
```suggestion
 * Represents the {@code EIP712Domain} part of an EIP-712 compliant data structure.
```

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712AppOrder.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                getMessage().getVolume(),
+                getMessage().getTag(),
+                getMessage().getDatasetrestrict(),
+                getMessage().getWorkerpoolrestrict(),
+                getMessage().getRequesterrestrict(),
+                getMessage().getSalt()
+        );
+    }
+
+}
```
Comment (mcornaton) [inline src/main/java/com/iexec/commons/poco/eip712/entity/EIP712AppOrder.java:69]:
```suggestion
}

```

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712RequestOrder.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                getMessage().getCategory(),
+                getMessage().getTrust(),
+                getMessage().getBeneficiary(),
+                getMessage().getCallback(),
+                getMessage().getParams(),
+                getMessage().getSalt()
+        );
+    }
+
+}
```
Comment (mcornaton) [inline src/main/java/com/iexec/commons/poco/eip712/entity/EIP712RequestOrder.java:83]:
```suggestion
}

```

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712WorkerpoolOrder.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                getMessage().getCategory(),
+                getMessage().getTrust(),
+                getMessage().getApprestrict(),
+                getMessage().getDatasetrestrict(),
+                getMessage().getRequesterrestrict(),
+                getMessage().getSalt()
+        );
+    }
+
+}
```
Comment (mcornaton) [inline src/main/java/com/iexec/commons/poco/eip712/entity/EIP712WorkerpoolOrder.java:73]:
```suggestion
}

```

File: src/test/resources/wallet.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "kdfparams": {
+      "salt": "68da8c2da545886bb5598a5a03bbfdf71ce580d927628c2b36b4055e790c0d75",
+      "n": 131072,
+      "dklen": 32,
+      "p": 1,
+      "r": 8
+    },
+    "mac": "4ac9818afdf43f9e5fc3b71665b59115f7e0b4bce8e82f631481bdb63e3847d3"
+  }
+}
```
Comment (mcornaton) [inline src/test/resources/wallet.json:21]:
```suggestion
}

```

Comment (jbern0rd) [reply src/test/resources/wallet.json]: Thanks, please see 441599d

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712WorkerpoolOrder.java (hunk 4 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/eip712/entity/EIP712WorkerpoolOrder.java]: Thanks, please see 441599d

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712RequestOrder.java (hunk 3 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/eip712/entity/EIP712RequestOrder.java]: Thanks, please see 441599d

File: src/main/java/com/iexec/commons/poco/eip712/entity/EIP712AppOrder.java (hunk 2 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/commons/poco/eip712/entity/EIP712AppOrder.java]: Thanks, please see 441599d

**Review by mcornaton — APPROVED** [review]
Thanks!

**Pull request merged**